base/ltxref.dtx

```diff
@@ -465,7 +466,7 @@
 \def\refstepcounter#1{%
   \UseSocket{refstepcounter}{%
     \stepcounter{#1}%
-    \edef\@currentcounter{#1}%
+    \edef\@currentcounter{\ifx*#1\@currentcounter\else#1\fi}%
```

Comment:
What if `#1` contains multiple tokens and starts with `*`? For example `\refstepcounter{*ctr}`.

Do we want to check that if `#1` is a single token here?

Reply:
I think one has to (such "*..." counter exist)

Reply:
I suppose so, I wanted to avoid doing too much overhead but i guess I should make this safer

Reply:
also `\@kernel@refstepcounter` needs the same

Reply:
```
\def\ltx@star@counter{*}
\def\refstepcounter#1{%
  \UseSocket{refstepcounter}{%
    \stepcounter{#1}%
    \edef\reserved@a{#1}%
    \ifx\reserved@a\ltx@star@counter\else
      \edef\@currentcounter{#1}%
    \fi
 .....
```

?

Reply:
oh I meant to do `\let\@currentcounter\reserved@a` inside the if

Reply:
resolved by 4cc48dfef I hope

Reply:
I think that'll do